File: api/app/schemas/booking.py
from __future__ import annotations

import uuid
from datetime import date, datetime

from pydantic import BaseModel


class CreateBookingRequest(BaseModel):
    flight_id: uuid.UUID
    seat_id: uuid.UUID


class BookingResponse(BaseModel):
    booking_id: uuid.UUID
    status: str
    seat_code: str
    flight_number: str
    gate: str
    departure_date: date
    departure_time: str
    booked_at: datetime


class BookingListResponse(BaseModel):
    items: list[BookingResponse]
    total: int

File: client/src/types/api/booking.ts
export type CreateBookingRequest = {
  flight_id: string
  seat_id: string
}

export type BookingResponse = {
  booking_id: string
  status: string
  seat_code: string
  flight_number: string
  gate: string
  departure_date: string
  departure_time: string
  booked_at: string
}

export type BookingListResponse = {
  items: BookingResponse[]
  total: number
}
